### ISSUE TITLE
FileNotFoundError when bem folder exists but surfaces were not created properly

### ISSUE THREAD

**Issue by MerlinDumeur** [mne-tools/mne-bids-pipeline#304]
Hi,

I've had a first run of the source pipeline crash (due not having the dev version of mne-bids), and upon restarting the command it crashed in `02-make_forward.py` with the following traceback:
```
  File "/home/merlin/PhD/mne-bids-pipeline/scripts/source/02-make_forward.py", line 99, in run_forward
    bem_model = mne.make_bem_model(subject=fs_subject,
  File "<decorator-gen-55>", line 24, in make_bem_model
  File "/home/merlin/miniconda3/envs/mne-bids/lib/python3.9/site-packages/mne/bem.py", line 578, in make_bem_model
    surfaces = _surfaces_to_bem(surfaces, ids, conductivity, ico)
  File "/home/merlin/miniconda3/envs/mne-bids/lib/python3.9/site-packages/mne/bem.py", line 488, in _surfaces_to_bem
    surfs[si] = read_surface(surf, return_dict=True)[-1]
  File "<decorator-gen-48>", line 24, in read_surface
  File "/home/merlin/miniconda3/envs/mne-bids/lib/python3.9/site-packages/mne/surface.py", line 718, in read_surface
    fname = _check_fname(fname, 'read', True)
  File "/home/merlin/miniconda3/envs/mne-bids/lib/python3.9/site-packages/mne/utils/check.py", line 178, in _check_fname
    raise FileNotFoundError(f'{name} does not exist: {fname}')
FileNotFoundError: File does not exist: /home/merlin/hMT+/derivatives/freesurfer/subjects/sub-01/bem/inner_skull.surf
```

Of course I just had to delete the `derivatives/freesurfer/subjects/sub-01/bem/` folder to force the pipeline to create all the correct files, however the cause of the error, and the fix might not be obvious for newcomers.

**Comment by hoechenberger** [comment]
We actually have a `recreate_bem` setting, allowing you to force-recreate the BEM model. By default, we reuse the existing file(s) to save computation time.

Any suggestions on how to improve this behavior are welcome!

**Comment by MerlinDumeur** [comment]
That's great, then probably just catching the `FileNotFoundError` when calling `mne.make_bem_model` with a message indicating that passing `recreate_bem=True` could help fix the issue.

I could open a PR for that if you'd like

**Comment by hoechenberger** [comment]
> I could open a PR for that if you'd like

Yes, please do give it a shot! :)

**Comment by MerlinDumeur** [comment]
> Actually … if we do catch a FileNotFoundError anyway … we might as well start recreating the BEM right away, WDYT? Because this indicates that it's in fact not there, so we wouldn't actually be recreate it, but create a proper BEM instead

It would be convenient, however it could also be suprising behavior. Say if only one file is missing, and I changed for whatever reason my config parameters, I would like to have the opportunity to check my config before recreating the BEM.

I think simply saying to the user that the BEM is incomplete and needs to be redone is probably better.

**Comment by hoechenberger** [comment]
> I think simply saying to the user that the BEM is incomplete and needs to be redone is probably better.

Ok, go for it! :)

**Comment by hoechenberger** [comment]
I believe this can be closed now that #309 has been merged.
